include/ttmlir/Dialect/TTMetal/Pipelines/TTMetalPipelines.h

```diff
@@ -12,8 +12,7 @@ namespace mlir::tt::ttmetal {
 //
 struct TTIRToTTMetalBackendPipelineOptions
     : public PassPipelineOptions<TTIRToTTMetalBackendPipelineOptions> {
-  ListOption<int64_t> meshShape{
-      *this, "mesh-shape", llvm::cl::desc("Set the multi-device mesh shape.")};
+  ::llvm::SmallVector<int64_t> meshShape;
```

Comment:
Why did this change?  I think we still want to be able to override mesh shape from the command line.

Reply:
I get the following error after the upgrading LLVM version:

```
/home/uaziz/tt-mlir/lib/Dialect/TTMetal/Pipelines/TTMetalPipelines.cpp:21:35: error: no viable overloaded '='
  implicitDeviceOptions.meshShape = options.meshShape;
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~
/opt/ttmlir-toolchain/include/llvm/ADT/SmallVector.h:1241:16: note: candidate function not viable: no known conversion from 'const ListOption<int64_t>' (aka 'const ListOption<long>') to 'const llvm::SmallVector<long, 6>' for 1st argument
  SmallVector &operator=(const SmallVector &RHS) {
```

If we want to keep ListOption structure, I can change it to copy the vector without using assignment operator, does that work?

`std::copy(options.meshShape.begin(), options.meshShape.end(), std::back_inserter(implicitDeviceOptions.meshShape));`

Reply:
No it fails with the same error: 

```
/home/uaziz/tt-mlir/lib/Dialect/TTMetal/Pipelines/TTMetalPipelines.cpp:21:35: error: no viable overloaded '='
  implicitDeviceOptions.meshShape = *options.meshShape;
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~
/opt/ttmlir-toolchain/include/llvm/ADT/SmallVector.h:1241:16: note: candidate function not viable: no known conversion from 'ArrayRef<long>' to 'const llvm::SmallVector<long, 6>' for 1st argument
  SmallVector &operator=(const SmallVector &RHS) {
```

However the following works: 

`implicitDeviceOptions.meshShape = ::llvm::SmallVector<int64_t>(options.meshShape.begin(), options.meshShape.end());`

Reply:
I'm fine with that or:
```
implicitDeviceOptions.meshShape = ::llvm::SmallVector<int64_t>(*options.meshShape);
```

Reply:
Doesn't work as well with the existing constructors:

```
/opt/ttmlir-toolchain/include/mlir/Pass/PassOptions.h:291:14: error: no viable conversion from 'const mlir::detail::PassOptions::ListOption<long, llvm::cl::parser<long>>' to 'const std::vector<long>'
      return static_cast<const std::vector<DataType> &>(*this);
             ^                                          ~~~~~
/home/uaziz/tt-mlir/lib/Dialect/TTMetal/Pipelines/TTMetalPipelines.cpp:21:66: note: in instantiation of member function 'mlir::detail::PassOptions::ListOption<long, llvm::cl::parser<long>>::operator*' requested here
  implicitDeviceOptions.meshShape = ::llvm::SmallVector<int64_t>(*options.meshShape);
```

Reply:
Did you make this change? It doesn't seem reflected on the review

Reply:
We want this to remain `ListOption<int64_t>`

Reply:
Updated it now with the change.